llms/mlx_lm/models/base.py

```diff
@@ -42,13 +42,13 @@ def create_causal_mask(
     return mask * -1e9
 
 
-def create_attention_mask(h: mx.array, cache: Optional[Any] = None):
+def create_attention_mask(h: mx.array, cache: Optional[Any] = None, layer_idx: int = 0):
     T = h.shape[1]
     if T > 1:
         window_size = None
         offset = 0
-        if cache is not None and cache[0] is not None:
-            c = cache[0]
+        if cache is not None and cache[layer_idx] is not None:
+            c = cache[layer_idx]
```

Comment:
Given below comment we can remove `layer_idx` from this API.. which I'd like to avoid making more complex since it's already a pretty leaky function.

Reply:
Sure thing!